browser/ui/commander/commander_service.cc

```diff
@@ -175,12 +176,24 @@ void CommanderService::UpdateTextFromCurrentBrowserOmnibox() {
 
 void CommanderService::UpdateText(const std::u16string& text, bool force) {
   auto* browser = chrome::FindLastActiveWithProfile(profile_);
-  if (!base::StartsWith(text, kCommandPrefix)) {
+  if (!browser) {
+    return;
+  }
+
+  auto has_prefix = base::StartsWith(text, kCommandPrefix);
+  if (!has_prefix && !browser->profile()->GetPrefs()->GetBoolean(
+                         omnibox::kCommanderSuggestionsEnabled)) {
+    return;
+  }
+
+  if (!has_prefix && text.empty()) {
```

Comment:
I think this could be simplified.
If text is empty, has_prefix is always false.
```suggestion
  if (text.empty()) {
```

Reply:
Nice catch!